Update Lite Scaffold

tanhengyeow opened the pull request:
Fixes #1 

CRA Lite's UI:
![image](https://user-images.githubusercontent.com/25320782/61350139-1e1a1800-a899-11e9-8279-5ee9c076ca2e.png)

tanhengyeow committed:
feat: update template

tanhengyeow committed:
chore: Add headerMenuConfig, update config files

tanhengyeow commented on scaffolds/Lite/jsconfig.json:
```diff
@@ -1,8 +1,8 @@
 {
   "compilerOptions": {
-    "baseUrl": ".",
+    "baseUrl": "./src",
```
We can only point to `./src`as the project's `baseUrl` can only be set to `src` or `node_modules`. Create React App does not support other values at this time. Refer [here](https://github.com/facebook/create-react-app/issues/6850).

imsobear replied:
But when we add page in iceworks, it will generate some code like `import PageA from '@/PageA';`

tanhengyeow replied:
As discussed, `baseAdapter.Router` in `iceworks` will be handling the @ alias prefix in import paths.

tanhengyeow commented on scaffolds/Lite/package.json:
```diff
@@ -5,50 +5,38 @@
-    "screenshot": "screenshot -l"
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "lint": "node_modules/eslint/bin/eslint.js src",
```
`react-scripts` does not support the `react-scripts lint` command. References:
1. [Run Lint from the command line?](https://github.com/facebook/create-react-app/issues/1217)
2. [added react-scripts lint #2729](https://github.com/facebook/create-react-app/pull/2729)

imsobear replied:
just `eslint .js,.jsx ./`, npm will find in node_modules first.

tanhengyeow reviewed (commented):
Ready for review. 

Setup project:
1. Run `npm install` in `cra-materials/scaffolds/Lite`.
2. Go to `iceworks` and open this project. The `Layout`, `Page`, `Router` and `Menu` directory structure should be shown correctly.
3. Tasks in the `Engineering` tab, `dev`, `build` and `lint` should work.

imsobear commented on scaffolds/Lite/package.json:
```diff
@@ -5,50 +5,38 @@
   "dependencies": {
     "@alifd/next": "^1.10.1",
-    "@icedesign/theme": "^1.x",
+    "node-sass": "^4.12.0",
```
Don't need node-sass and `@alife/next`

tanhengyeow replied:
The `PageLoading` component [depends](https://github.com/ice-lab/cra-materials/blob/master/scaffolds/Lite/src/components/PageLoading/index.jsx#L2) on `@alifd/next` and [router.jsx](https://github.com/ice-lab/cra-materials/blob/master/scaffolds/Lite/src/router.jsx#L42) uses `PageLoading` component.

imsobear replied:
div is ok, just simple loading

imsobear commented on scaffolds/Lite/package.json:
```diff
@@ -5,50 +5,38 @@
-    "ice-plugin-moment-locales": "^0.1.0",
-    "ice-scripts": "^2.0.0"
+    "react-router-dom": "^5.0.1",
+    "react-scripts": "3.0.1"
```
react-scripts maybe in devDependencies? and why locked version to 3.0.1

tanhengyeow replied:
A default CRA app has `react-scripts` as a dependency instead of a devDependency. I think this is intentional, refer [here](https://github.com/facebook/create-react-app/issues/4617).

imsobear commented on scaffolds/Lite/src/index.js:
```diff
@@ -1,12 +1,5 @@
 import ReactDOM from 'react-dom';
 import router from './router';
-
 import './global.scss';
```
maybe use css?

imsobear commented on scaffolds/Lite/src/router.jsx:
```diff
@@ -1,8 +1,8 @@
-import routes from '@/config/routes';
-import PageLoading from '@/components/PageLoading';
+import routes from 'config/routes';
+import PageLoading from 'components/PageLoading';
```
Looks like have not serviceWorker.js file?

tanhengyeow replied:
Yes I didn't import that file from a CRA app, do we want that?

imsobear replied:
serviceWorker.js looks like a symbol of CRA.

temper357 commented on scaffolds/Lite/package.json:
```diff
@@ -62,5 +50,8 @@
     "url": "https://github.com/ice-lab/cra-materials/tree/master/scaffolds/Lite"
+  },
+  "devDependencies": {
+    "moment": "^2.24.0"
```
`moment` is also not needed.

tanhengyeow committed:
chore: Update PR

- Update package.json scripts config and remove unnecessary dependencies
- Use css instead of scss
- Improve css styles
- Fix bug in anchor tags of the Header component
- Fix other review comments

tanhengyeow committed:
chore: Update PR

- Admin for package.json
- Add Service Worker files to scaffold

tanhengyeow committed:
Remove unnecessary style

tanhengyeow commented on scaffolds/Lite/.eslintrc:
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./node_modules/eslint-config-react-app/index.js",
+  "rules"  : {
+    // TODO
```
What type of default ESLint rules should we include in this scaffold?

tanhengyeow reviewed (commented):
@imsobear @temper357 Ready for review again 😄

temper357 committed:
feat: add depenedency

temper357 committed:
chore: add upload scripts

temper357 committed:
chore: add dependency
